### ISSUE TITLE
Stopping subscription before it's ready can leave observers running forever

### ISSUE THREAD

**Issue by etyp** [peerlibrary/meteor-reactive-publish#37]
### The issue
I've found an edge case when using this package alongside [meteorhacks:unblock's publication unblock](https://github.com/meteorhacks/unblock) where stopping a subscription before it's ready on the client will leave reactive `find`s inside publish autoruns open forever, even after a session is closed.

### To reproduce
To reproduce, you'll need a publication handle which uses both `this.unblock()` and `this.autorun()` like so:

```
// Where: Items = Meteor collection
Meteor.publish('items', function () {
  this.unblock();
  this.autorun(() => {
    // Some arbitrary findOne/fetch/map
    OtherCollection.findOne();
    return Items.find({});
  });
});
```

Then on the client, if you subscribe to the publication but stop it before it's ready like:

```
const handle = Meteor.subscribe('items');
handle.stop();
```

The `Items` cursor will stop as expected, but the `OtherCollection` cursor will exists **forever, even if the user closes out their session**. This issue in tandem with the issue where [finds on autoruns always use polling driver](https://github.com/peerlibrary/meteor-reactive-mongo/issues/3) can lead to rough performance on servers even after users close out their sessions.

### Temporary workaround
For now, you can keep track of a publication's computations and stop them inside of a `publish.onStop` callback. This will ensure that any reactive queries in the autorun are stopped:

```
Meteor.publish('items', function () {
  this._trackedComputations = [];
  this.onStop(() => {
    while (this._trackedComputations.length) {
      const c = this._trackedComputations.shift();
      c.stop();
    }
  });
  this.unblock();
  this.autorun((computation) => {
     this._trackedComputations.push(computation);
    // Some arbitrary findOne/fetch/map
    OtherCollection.findOne();
    return Items.find({});
  });
});
```

**Comment by etyp** [comment]
This isn't super common in our app, but we do have 2 cases where subscriptions are stopped before ready because the UI component they're registered in invalidates and stops the subscription really quickly.

It's avoidable, but I'd rather have a safety valve built in on this package.

I've submitted a PR #38 with an approach I tested and can confirm fixes this. I'm not sure if there's a better approach though. Let me know your thoughts @mitar :)

And thanks again for such a seriously amazing set of packages!
